### ISSUE TITLE
Questions about using reinforcement learning training

### ISSUE THREAD

**Issue by King9299** [rte-france/Grid2Op#539]
Now I am using reinforcement learning to conduct related research in Grid2op environment (version: 1.9.5), Grid2op environment is a very perfect environment, which can bring good help to my research. However, when I was training with DQN/PPO/SAC algorithm, I had some questions that I could not understand. I would be grateful if you could help me.

***********  Set an Environment **********************************************************************
bk_cls = LightSimBackend
env_name = "l2rpn_case14_sandbox"
take_env = grid2op.make(env_name, reward_class=IncreasingFlatReward, backend=LightSimBackend(),other_rewards={'loss': LossReward})

take_env.deactivate_forecast()
take_env.parameters.NB_TIMESTEP_OVERFLOW_ALLOWED = 3
take_env.parameters.NB_TIMESTEP_RECONNECTION = 12
take_env.parameters.NB_TIMESTEP_COOLDOWN_LINE = 3
take_env.parameters.NB_TIMESTEP_COOLDOWN_SUB = 3
take_env.parameters.HARD_OVERFLOW_THRESHOLD = 200.0

env = GymEnv(take_env)

env.action_space = DiscreteActSpace(env.init_env.action_space, attr_to_keep=["set_bus","set_line_status","redispatch"])
env.observation_space = BoxGymObsSpace(env.init_env.observation_space,attr_to_keep=["v_or","v_ex","a_or","a_ex","rho","topo_vect","timestep_overflow"])
**********************************************************************************************************

Q : I'm in an environment called "l2rpn_case14_sandbox", ActSpace is defined with (DiscreteActSpace(env.init_env.action_space, attr_to_keep=["set_bus","set_line_status","redispatch"])). The algorithm can then start training smoothly. But after training for a short period of time, always output the same action value, which I do not understand. 

I want to know what kind of action a good agent should output.

**Comment by BDonnot** [comment]
Hello

Before I answer your question a quick remark.

This has absolutely no effect:
```python
take_env.parameters.NB_TIMESTEP_OVERFLOW_ALLOWED = 3
take_env.parameters.NB_TIMESTEP_RECONNECTION = 12
take_env.parameters.NB_TIMESTEP_COOLDOWN_LINE = 3
take_env.parameters.NB_TIMESTEP_COOLDOWN_SUB = 3
take_env.parameters.HARD_OVERFLOW_THRESHOLD = 200.0
```
see the detailed explanation here
https://grid2op.readthedocs.io/en/latest/environment.html#id0

If you need to change the parameters use: 
https://grid2op.readthedocs.io/en/latest/environment.html#grid2op.Environment.BaseEnv.change_parameters


Now to answer your question:

> But after training for a short period of time, always output the same action value, which I do not understand.

It's probably an issue with your RL agent. For example have you tried to tune its hyper parameters ? Learning rate and size of the neural network might help. Also you can start your tuning from already working parameters as found here https://l2rpn-baselines.readthedocs.io/en/latest/external_contributions.html with contribution of some previous L2RPN winners.

> I want to know what kind of action a good agent should output.

This is precisely what needs to be found by your agent :-) A good starting place for this are the papers about "l2rpn" and the l2rpn-baselines repository.

**Comment by King9299** [comment]
Thanks for your reply, I have my research target for the next stage

**Comment by King9299** [comment]
Is it reasonable to output the same action in adjacent time steps?
action_space : Dis(21)---------("change_line_status")

Ex : eposide 1 :  action [5,5,5,5,5,10,5,5,5,5,5,5,20,5,5,5,5,5,5,2,6,7]
eposide 2 :  action [15,15,15,15,15,15,15,15,15,15,15,15,20,15,15,15,15,15,6,12,15,7]

**Comment by BDonnot** [comment]
Hello,

 don't know what you mean by "reasonable" here so I can't really answer the question.

Your agent seems to change the line status, so yes it's "reasonable": at one step it wants to connect the powerline, at the other step it wants to disconnect it etc. It makes "perfect sense".
Whether that's realistic, most definitely not. Your agent should chose on whether it wants the powerline connected or not and this should last for a few steps at least.

**Comment by BDonnot** [comment]
I am closing this issue as you appear to be satisfied with my last answer. If not let me know and feel free to re open it :-)
